tests/test_runhistory/test_runhistory_encoder.py

```diff
@@ -39,70 +41,127 @@ def test_transform(runhistory, make_scenario, configspace_small, configs):
     )
 
     # Normal encoder
-    encoder = RunHistoryEncoder(scenario=scenario, considered_states=[StatusType.SUCCESS])
+    encoder = RunHistoryEncoder(
+        scenario=scenario, considered_states=[StatusType.SUCCESS]
+    )
     encoder.runhistory = runhistory
+
+    # TODO: Please replace with the more general solution in the future
+    # upper = np.array([hp.upper_vectorized for hp in space.values()])
+    # lower = np.array([hp.lower_vectorized for hp in space.values()])
+    # -
+    # Categoricals are upperbounded by their size, rest of hyperparameters are
+    # upperbounded by 1.
+    upper_bounds = {
+        hp.name: (hp.get_size() - 1)
+        if isinstance(hp, CategoricalHyperparameter)
+        else 1.0
+        for hp in configspace_small.get_hyperparameters()
+    }
+    # Need to ensure they match the order in the Configuration vectorized form
+    sorted_by_indices = sorted(
+        upper_bounds.items(),
+        key=lambda x: configspace_small._hyperparameter_idx[x[0]],
+    )
+    upper = np.array([upper_bound for _, upper_bound in sorted_by_indices])
+    lower = 0.0
```

Comment:
I guess the lower bound is always 0 no matter the type of HP?

Reply:
Yes, but theoretically someone can implement their own HP where that's not the case. Hence the change in the above `#TODO`